wgpu/src/lib.rs

```diff
@@ -1936,12 +1958,29 @@ impl Instance {
     /// - On macOS/Metal: will panic if not called on the main thread.
     /// - On web: will panic if the `raw_window_handle` does not properly refer to a
     ///   canvas element.
-    pub unsafe fn create_surface<
-        W: raw_window_handle::HasWindowHandle + raw_window_handle::HasDisplayHandle,
+    pub fn create_surface<
+        'window,
+        W: 'window + HasWindowHandle + HasDisplayHandle + WasmNotSend + WasmNotSync,
     >(
+        &self,
+        window: W,
+    ) -> Result<Surface<'window>, CreateSurfaceError> {
+        let mut surface = unsafe { self.create_surface_from_raw(&window) }?;
+        surface._surface = Some(Box::new(window));
+        Ok(surface)
+    }
+
+    /// See [`create_surface()`](Self::create_surface) for more details.
```

Comment:
Could we expand on the differences between the above and this. This should explain why this has a bunch of requirements the other doesn't?